Public/Src/Engine/Scheduler/Scheduler.cs

```diff
@@ -289,6 +289,19 @@ public partial class Scheduler : IPipScheduler, IPipExecutionEnvironment, IFileC
         private int m_maxUnresponsivenessFactor = 0;
         private DateTime m_statusLastCollected = DateTime.MaxValue;
 
+        /// <summary>
+        /// Output file contents which affected by the source change of this build. 
+        /// </summary>
+        /// <remarks>
+        /// This list should be initialized with the source change list provided by GBR.
+        /// </remarks>
+        private readonly ConcurrentBigSet<AbsolutePath> m_sourceChangeAffectedOutputFiles = new ConcurrentBigSet<AbsolutePath>();
+
+        /// <summary>
+        /// Output file contents which affected by the source change of this build. 
+        /// </summary>
+        private readonly ConcurrentBigSet<AbsolutePath> m_sourceChangeAffectedOutputDirectories = new ConcurrentBigSet<AbsolutePath>();

```

Comment:
>AbsolutePath [](start = 42, length = 12)

Should `AbsolutePath` be `DirectoryArtifact`? Also, can the set be joined with `m_sourceChangeAffectedOutputFiles` ?

Reply:
I still think AbsolutePath is enough.
The reason I seperate the set to sourceChangeAffectedOutputFiles and sourceChangeAffectedOutputFiles is that the computing algorithm interact with directoryDenpendencies and Denpendencies as well as dynamicallyObservedFiles and dynamicallyObservedEnumerations. Making it two seperate set will make the computation algorithm easier.